### ISSUE TITLE
[Data Grid] Column groups - header hierarchy is announced incorrectly NVDA screen reader

### ISSUE THREAD

**Issue by tweiss2** [mui/mui-x#13374]
### Steps to reproduce

Link to live example: [(required)](https://codesandbox.io/p/sandbox/fast-framework-9k6kdm?file=%2Fsrc%2FDemo.tsx)
https://codesandbox.io/p/sandbox/fast-framework-9k6kdm?file=%2Fsrc%2FDemo.tsx

### Current behavior

Focus on ID column header. Notice screen reader reads 
`ID column header sorted ascending row 2 Internal (not freeReordering)
ID column 2`

using the arrow keys move to focus on column header `is admin`
notice the screen reader reads:
`is admin column header row 2 Full Name (freeReordering)
is admin column 3`

### Expected behavior

When focusing on the `is admin` column header. I would have expected the screen reader to read the correct column hierarchy . `is admin` is a child of the `Internal (not freeReordering)` column and not `Full Name (freeReordering)` column header.

`is admin column header row 2 Internal (not freeReordering)
is admin column 3`

### Context

header hierarchy is announced incorrectly
The screen reader should provide full context to end use to the current column header, and not announce an incorrect column header.

### Your environment

browser: chrome
NVDA speech
Windows machine 

**Search keywords**: Column group screen reader reading incorrect column hierarchy
**Order ID**: 71872

**Comment by tweiss2** [comment]
Expected behavior may also be to just read the focused column header and not all headers in this spot. 
For example: when focusing on the `is admin` column header, the screen reader should read-
`is admin column header row 2 is admin column 3`

**Comment by oliviertassinari** [comment]
It's an NVDA bug no https://adrianroselli.com/2023/02/avoid-spanning-table-headers.html?

**Comment by michelengelen** [comment]
> https://adrianroselli.com/2023/02/avoid-spanning-table-headers.html

Looks very much like it.

Is there anything we could do to still support it?
Might be nearly impossible though.
